pkg/phlaredb/block_querier.go

```diff
@@ -905,6 +938,21 @@ func (b *singleBlockQuerier) SelectMatchingProfiles(ctx context.Context, params
 	return iter.NewMergeIterator(maxBlockProfile, false, iters...), nil
 }
 
+func (b *singleBlockQuerier) Series(ctx context.Context, params *ingestv1.SeriesRequest) ([]*typesv1.Labels, error) {
+	sp, ctx := opentracing.StartSpanFromContext(ctx, "Series Block")
+	defer sp.Finish()
+
+	// TODO(bryan) do we need to close this?
+	err := b.Open(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO(bryan) get labels from block and filter by matchers/label_names
```

Comment:
Not exactly sure how to get label pairs from a block. I was following how it was queried from the ingester, but there is some differences in the head block format and a block that's been flushed to storage.

I was browsing `profiles.parquet` because it had a `labels` column, but none of the rows had any labels.

Reply:
As per our slack discussion, there two different kind of labels, in this case we are interested in the external labels, which can be found in `index.tsdb` and are mappend into profiles.tsdb over the series index